apps/condo/domains/banking/access/BankAccount.js

```diff
@@ -3,21 +3,45 @@
  */
 const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
 
+const { checkBankIntegrationsAccessRights } = require('@condo/domains/banking/utils/accessSchema')
 const {
     queryOrganizationEmployeeFor,
     queryOrganizationEmployeeFromRelatedOrganizationFor,
 } = require('@condo/domains/organization/utils/accessSchema')
+const { SERVICE } = require('@condo/domains/user/constants/common')
+
+const { BANK_INTEGRATION_IDS } = require('../constants')
 
 /**
  * BankAccount entity can be read either by:
  * 1. By admin or support
  * 2. Organization employee
  */
-async function canReadBankAccounts ({ authentication: { item: user } }) {
+async function canReadBankAccounts ({ authentication: { item: user }, context }) {
     if (!user) return throwAuthenticationError()
     if (user.deletedAt) return false
     if (user.isAdmin || user.isSupport) return {}
 
+    if (user.type === SERVICE) {
+        if (await checkBankIntegrationsAccessRights(context, user.id, [BANK_INTEGRATION_IDS.SBBOL])) return {
+            OR: [{
+                integrationContext: {
+                    deletedAt: null,
+                    integration: {
+                        deletedAt: null,
+                        id: BANK_INTEGRATION_IDS.SBBOL,
+                    },
+                },
+            },
+            {
+                integrationContext_is_null: true,
+            }],
+
+        }
+
+        return false
```

Comment:
It seems like `await checkBankIntegrationsAccessRights` is redundant here, because you can specify filtering condition like this:
```suggestion
        return {
            OR: [{
                integrationContext: {
                    deletedAt: null,
                    integration: {
                        deletedAt: null,
                        accessRights_some: { user: { id: user.id }, deletedAt: null }, deletedAt: null,
                    },
                },
            },
            {
                integrationContext_is_null: true,
            }],

        }

        return false
```